src/figuresClasses.ts

```diff
@@ -1,11 +1,75 @@
-export interface Figure {}
+type Shape = 'triangle' | 'circle' | 'rectangle';
+type Color = 'red' | 'green' | 'blue';
 
-export class Triangle implements Figure {}
+export interface Figure {
+  shape: Shape;
+  color: Color;
+  getArea(): number;
+}
+
+export class Triangle implements Figure {
+  shape: Shape = 'triangle';
+
+  constructor(
+    public color: Color,
+    private a: number,
+    private b: number,
+    private c: number,
+  ) {
+    if (a <= 0 || b <= 0 || c <= 0) {
+      throw new Error('All sides length must by > 0');
+    }
+
+    if (Math.max(a, b, c) >= a + b + c - Math.max(a, b, c)) {
+      throw new Error(`sides ${a}, ${b} and ${c} can't form a triangle`);
+    }
+  }
+
+  getArea(): number {
+    const s = (this.a + this.b + this.c) / 2;
+    const A = Math.sqrt(s * (s - this.a) * (s - this.b) * (s - this.c));
+
+    return Math.round(A * 100) / 100;
+  }
+}
 
-export class Circle implements Figure {}
+export class Circle implements Figure {
+  shape: Shape = 'circle';
 
-export class Rectangle implements Figure {}
+  constructor(
+    public color: Color,
+    private radius: number,
+  ) {
+    if (radius <= 0) {
+      throw new Error('Radius length must by > 0');
+    }
+  }
+
+  getArea(): number {
+    const result = Math.PI * Math.pow(this.radius, 2);
+
+    return Math.floor(result * 100) / 100;
+  }
+}
+
+export class Rectangle implements Figure {
+  shape: Shape = 'rectangle';
+
+  constructor(
+    public color: Color,
+    private width: number,
+    private height: number,
+  ) {
+    if (width <= 0 || height <= 0) {
+      throw new Error('All sides length must by > 0');
```

Comment:
[CODE STYLE] - Make sure your error messages are detailed. Instead of 'All sides length must by > 0', it could be more descriptive, like 'Width and height must be greater than 0'.